clang/lib/StaticAnalyzer/Core/ExprEngine.cpp

```diff
@@ -3742,23 +3742,20 @@ void ExprEngine::evalLocation(ExplodedNodeSet &Dst,
   BldrTop.addNodes(Tmp);
 }
 
-std::pair<const ProgramPointTag *, const ProgramPointTag*>
-ExprEngine::geteagerlyAssumeBinOpBifurcationTags() {
-  static SimpleProgramPointTag
-         eagerlyAssumeBinOpBifurcationTrue(TagProviderName,
-                                           "Eagerly Assume True"),
-         eagerlyAssumeBinOpBifurcationFalse(TagProviderName,
-                                            "Eagerly Assume False");
-  return std::make_pair(&eagerlyAssumeBinOpBifurcationTrue,
-                        &eagerlyAssumeBinOpBifurcationFalse);
+std::pair<const ProgramPointTag *, const ProgramPointTag *>
+ExprEngine::getEagerlyAssumeBifurcationTags() {
+  static SimpleProgramPointTag TrueTag(TagProviderName, "Eagerly Assume True"),
+      FalseTag(TagProviderName, "Eagerly Assume False");
+
+  return std::make_pair(&TrueTag, &FalseTag);
 }
 
-void ExprEngine::evalEagerlyAssumeBinOpBifurcation(ExplodedNodeSet &Dst,
-                                                   ExplodedNodeSet &Src,
-                                                   const Expr *Ex) {
+void ExprEngine::evalEagerlyAssumeBifurcation(ExplodedNodeSet &Dst,
+                                              ExplodedNodeSet &Src,
+                                              const Expr *Ex) {
   StmtNodeBuilder Bldr(Src, Dst, *currBldrCtx);
 
-  for (const auto Pred : Src) {
+  for (ExplodedNode *Pred : Src) {
```

Comment:
This small clarification is unrelated to everything else -- I just spelled out the type explicitly because that's the preferred style and I'm already cleaning up many things in this function.

In theory I could say `ExplodedNode * const Pred` but I don't think that it's worth to emphasize the `const`ness of the pointer.

Reply:
I guess initially it wanted to be `const auto *`. There is no need to make the pointer `const`. Making the pointed value `const` would make sense if it's applicable here.

Reply:
> Making the pointed value const would make sense if it's applicable here.

The pointed value cannot be made `const`, we're passing it to functions that modify it. (In general, `ExplodedNode`s are almost never `const` during the analysis, because we can add successor nodes to them in many situations and that modifies one of their data members, the list of successors.)